task/runner.go

```diff
@@ -58,26 +58,33 @@ func (r *Runner) runAsyncJob(job func(), wg *sync.WaitGroup) {
 // determines whether the task runner should run at all
 // tasks are to be run at most once every 24 hours
 // also, this is not to run in batch query mode
-func (r *Runner) shouldRun() bool {
+func (r *Runner) shouldRun() (returnValue bool) {
+	returnValue = true
+
+	defer func() {
```

Comment:
why defer this - why not just return out at once in batch mode?

also - not too keen on usage of named return value here - I think often it makes code more confusing. I guess we need it for defer... but could be removed if we remove defer?